src/hooks/use-axios.tsx

```diff
@@ -31,24 +35,37 @@ export const useAxios = () => {
     const handleError = (error: any) => {
       const status = error?.response?.status ?? error?.status
 
-      if (status === 401) {
-        toastLibrary.custom(
-          (t) => (
-            <div onClick={() => signIn()} className="cursor-pointer">
-              <ToastContent
-                title={"Unauthorized"}
-                description={
-                  "You may need to sign in again. Click here to sign in again"
-                }
-                variant={"destructive"}
-                t={t}
-              />
-            </div>
-          ),
-          {
-            position: "top-center",
-          },
-        )
+      if (status === 401 && session) {
+        // Clear the stale session from localStorage
+        setSession(null)
+
+        // Only show toast once to avoid spam
+        if (!has401ToastBeenShown) {
+          has401ToastBeenShown = true
+          toastLibrary.custom(
+            (t) => (
+              <div onClick={() => signIn()} className="cursor-pointer">
+                <ToastContent
+                  title={"Unauthorized"}
+                  description={
+                    "You may need to sign in again. Click here to sign in again"
+                  }
+                  variant={"destructive"}
+                  t={t}
+                />
+              </div>
+            ),
+            {
+              position: "top-center",
+              duration: 5000,
+            },
+          )
+
+          // Reset the flag after a delay so future 401s can show toast again
+          setTimeout(() => {
+            has401ToastBeenShown = false
+          }, 6000)
```

Comment:
Not sure if this is the correct pattern, cause all the components will be having the same `setTimeout` which might cause race conditions.